### PR TITLE
New data sources `azurerm_management_group_template_deployment`, `azurerm_resource_group_template_deployment`, `azurerm_subscription_template_deployment`, `azurerm_tenant_template_deployment`

### LINKED ISSUE #14523: Support for Azure Template Deployment data sources
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It would be useful to have data source equivalents for the following resources:
* `azurerm_tenant_template_deployment`
* `azurerm_subscription_template_deployment`
* `azurerm_resource_group_template_deployment`
* `azurerm_management_group_template_deployment`

This way we can leverage the outputs of these deployments in Terraform resources.

### New or Affected Resource(s)

* `data azurerm_tenant_template_deployment`
* `data azurerm_subscription_template_deployment`
* `data azurerm_resource_group_template_deployment`
* `data azurerm_management_group_template_deployment`

### Potential Terraform Configuration

```hcl
data "azurerm_resource_group_template_deployment" "example" {
  name                = "example-deploy"
  resource_group_name = "example-group"
}

output arm_example_output {
  value = jsondecode(data.azurerm_resource_group_template_deployment.example.output_content).exampleOutput.value
}
```

### PR DESCRIPTION
**Pull request by sdebruyn**
Fixes #14523

Adds data sources as mentioned in the title so that you can retrieve the values of outputs of deployed templates.

### CODE REVIEW

**Comment by sdebruyn** [comment]
I know docs are missing, will add those ASAP

**Comment by katbyte** [comment]
test failure @sdebruyn 
```
------- Stdout: -------
=== RUN   TestAccDataSourceManagementGroupTemplateDeployment
=== PAUSE TestAccDataSourceManagementGroupTemplateDeployment
=== CONT  TestAccDataSourceManagementGroupTemplateDeployment
    testing_new_config.go:81: no "id" found in attributes
    testing_new.go:53: no "id" found in attributes
--- FAIL: TestAccDataSourceManagementGroupTemplateDeployment (136.73s)
FAIL

```

**Comment by sdebruyn** [comment]
I have no idea how I can run these or what I should do to fix them. Where can I find some documentation on how the acceptance tests should run?

**Comment by sdebruyn** [comment]
sorry about that @katbyte, should be fixed now

**Comment by sdebruyn** [comment]
@katbyte @tombuildsstuff Any chance to merge this in before we get to 3.0?

**Comment by sdebruyn** [comment]
Rebased on latest main
@tombuildsstuff @katbyte Would you care to took a look please? The label says XL but most of the changes are very similar to what's already there.

File: internal/services/resource/management_group_template_deployment_data_source.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.TemplateDeploymentName,
+			},
+
+			"management_group_name": {
```
[the remark this answers is not in the document]
Comment (sdebruyn) [reply internal/services/resource/management_group_template_deployment_data_source.go]: fixed

File: internal/services/resource/management_group_template_deployment_data_source.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: mgValidate.ManagementGroupName,
```
[the remark this answers is not in the document]
Comment (sdebruyn) [reply internal/services/resource/management_group_template_deployment_data_source.go]: fixed

**Comment by sdebruyn** [comment]
@tombuildsstuff Thanks for the first review 🙌 It is now ready for re-review 😉

**Review by tombuildsstuff — APPROVED** [review]
hey @sdebruyn 

Thanks for pushing those changes - apologies for the delay on the re-review here!

Taking a look through this now LGTM 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
Acceptance Tests look good 👍

**Pull request merged**